### PR TITLE
Add new erosion strategy for /MAT/LAW124

### PR DESCRIPTION
**Pull request by vdavaze**
#### Checklist

- [x] The title of the PR is reviewed
- [x] There is no text before the checklist section
- [x] The following two sections are filled (or deleted)
- [x] This PR has been previewed 

#### Description of the feature or the bug

Initial element deletion method for recently implemented /MAT/LAW124 was to brittle in comparison with physical results

#### Description of the changes

Change the erosion strategy and add the option to deactivate element deletion.

### CODE REVIEW

File: engine/source/materials/mat/mat124/sigeps124.F [hunk 1]
```diff
@@ -52,8 +52,7 @@ SUBROUTINE SIGEPS124(
       !=======================================================================
       !      Dummy arguments
       !=======================================================================
-      INTEGER NEL,NUPARAM,NUVAR,INLOC,IPG,NPG,
```
Comment (sebastienVilleneuve) [inline engine/source/materials/mat/mat124/sigeps124.F:55]: INTENT statement is missing here

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
Hi @vdavaze 

some minor change to perform ( INTENT)

Thanks,
Sébastien

**Comment by vdavaze** [comment]
Hi, 

Sorry for this omission. Changes have been added. 

Thanks a lot. 

Valentin

**Review by sebastienVilleneuve — DISMISSED** [review]

**Comment by sebastienVilleneuve** [comment]
Thanks a lot @vdavaze 

Have a nice day,
Sébastien

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**